### DISCUSSION TITLE
Can i add leaflet in nuxt3? and if yes then can you provide me the steps for that

### DISCUSSION THREAD

**Discussion by NomanaReveation** [nuxt/nuxt#15989] [Questions]
I want to add leaflet map in my nuxt3 application . i tried https://www.npmjs.com/package/nuxt-leaflet these steps but it is giving me error  Error loading external "C:/eye-hospital/desiign/eye-hospital-project/Eye-Hospital/node_modules/leaflet/dist/leaflet-src.js". so can you plaese help with setup for this

**Comment by wobsoriano** [comment]
That package is for vue2. Try this one https://github.com/vue-leaflet/vue-leaflet

```ts
// nuxt.config.ts
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
  css: ['leaflet/dist/leaflet.css'],
});
```

```html
<script setup lang="ts">
import { LMap, LGeoJson } from '@vue-leaflet/vue-leaflet'

const geojson = {
  type: "FeatureCollection",
  features: [],
};

const geojsonOptions = {};
</script>

<template>
  <LMap style="height: 50vh">
    <LGeoJson :geojson="geojson" :options="geojsonOptions" />
  </LMap>
</template>

```

If you want to make the components global, you can do it like this:

```ts
// plugins/leaflet.ts
import { defineNuxtPlugin } from '#app'
import { LMap, LGeoJson } from '@vue-leaflet/vue-leaflet'

export default defineNuxtPlugin(nuxtApp => {
  nuxtApp.vueApp.component('LMap', LMap)
  nuxtApp.vueApp.component('LGeoJson', LGeoJson)
})
```

**Reply by x-N0** [reply]
Aside that we need to have the pkg `leaflet` installed, this is to use their css.

Also I'll recommend you to use [this](https://github.com/vue-leaflet/vue3-demo-project/blob/master/src/App.vue) as your template to test the map render when the configuration was done.

Hope it helps, guys.

**Reply by KristianApostolov** [reply]
This package does not work either..
And help would be greatly appreciated!

**Comment by riso348** [comment]
Hi there, i was not able to run the map because of error in Icon in leaflet/utile, so i try to disable default

Using package @vue-leaflet/vue-leaflet
```
<LMap ref="map" id="map" :zoom="zoomLevel" :center="mapCenter" :useGlobalLeaflet="false"
        :zoomAnimation="false" :fadeAnimation="false"
        :zoomAnimationThreshold="0"
        :noBlockingAnimations="true">
    <LTileLayer
        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        layer-type="base"
        name="OpenStreetMap"
    ></LTileLayer>
  </LMap>
```


Without the useGlobalLeaflet it doen not for me. You can either use local component such as i did 

```

<script setup lang="ts">
import "leaflet/dist/leaflet.css";
import {LMap, LTileLayer, LMarker} from "@vue-leaflet/vue-leaflet";
....
```

or make a plugin as you mentioned. Now i can see a map. Hope i help you :) 


.. But after a few minutes of playing with it i end up with only leaflet js library itsef without any other package..
